valuable/src/structable.rs

```diff
@@ -7,16 +7,103 @@ pub trait Structable: Valuable {
     fn definition(&self) -> StructDef<'_>;
 }
 
+/// Name, fields, and other struct-level information.
+///
+/// Returned by [`Structable::definition()`], `StructDef` provides the caller
+/// with information about the struct's definition.
+///
+/// [`Structable::definition()`]: Structable::definition
 #[non_exhaustive]
 pub enum StructDef<'a> {
+    /// The struct is statically-defined, all fields are known ahead of time.
```

Comment:
nit/tioli: maybe something like:
```suggestion
    /// The struct is statically-defined, all fields are known ahead of time.
    ///
    /// Most `Structable` definitions for Rust struct types will be `StructDef::Static`.
```